### PR TITLE
Torch issue716

### PR DESCRIPTION
**Pull request by edoaltamura**
### Summary
Merges fixes for issue 716 into main Dev. Closes https://github.com/OkuyanBoga/hc-qiskit-machine-learning/issues/16. Closes https://github.com/qiskit-community/qiskit-machine-learning/issues/716.

✅ I have added the tests to cover my changes.
✅ I have updated the documentation accordingly.
✅ I have read the CONTRIBUTING document.

### CODE REVIEW

File: qiskit_machine_learning/connectors/torch_connector.py [hunk 1]
```diff
@@ -52,12 +52,57 @@ class Module:  # type: ignore
...
+    if return_type == "input":
+        signature = f"{trace[:-1]},{trace:s}->{trace[0] + trace[2:]}"
+    elif return_type == "weight":
+        signature = f"{trace[:-1]},{trace:s}->{trace[-1]}"
+    else:
+        raise ValueError(
+            f'The only allowed return types are ["input", "weight"], got {return_type:s} instead.'
+        )
+
+    return signature
```
Comment (OkuyanBoga) [inline qiskit_machine_learning/connectors/torch_connector.py:96]: Is this covered by any unit tests?
- A test case for every `raise`
- Input/Output unit tests should be included

File: qiskit_machine_learning/connectors/torch_connector.py [hunk 2]
```diff
@@ -52,12 +52,57 @@ class Module:  # type: ignore
         pass
 
 
+def get_einsum_signature(n_dimensions: int, return_type: Literal["input", "weight"]) -> str:
```
Comment (OkuyanBoga) [inline qiskit_machine_learning/connectors/torch_connector.py:55]: Is this expected to be used by the users or is this an internal function?

**Review by OkuyanBoga — CHANGES_REQUESTED** [review]

File: qiskit_machine_learning/connectors/torch_connector.py (hunk 2 above)
Comment (edoaltamura) [reply qiskit_machine_learning/connectors/torch_connector.py:55]: Just internal - would you recommend renaming to `_get_einsum_signature`?

File: qiskit_machine_learning/connectors/torch_connector.py (hunk 1 above)
Comment (edoaltamura) [reply qiskit_machine_learning/connectors/torch_connector.py]: Added in https://github.com/OkuyanBoga/hc-qiskit-machine-learning/pull/18/commits/034785b1d3ff525633329445249a56abf704a2f3.

File: qiskit_machine_learning/connectors/torch_connector.py (hunk 2 above)
Comment (edoaltamura) [reply qiskit_machine_learning/connectors/torch_connector.py]: Renamed in https://github.com/OkuyanBoga/hc-qiskit-machine-learning/pull/18/commits/034785b1d3ff525633329445249a56abf704a2f3.

**Review by edoaltamura — COMMENTED** [review]
Fixed comments and ready for merge.

**Review by OkuyanBoga — APPROVED** [review]

**Comment by OkuyanBoga** [comment]
You can submit PR to the main repository now.

**Pull request merged**